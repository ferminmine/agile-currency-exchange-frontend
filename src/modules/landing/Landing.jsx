import React from 'react';
import { compose } from 'redux';
import withStyles from 'react-jss';
import PropTypes from 'prop-types';
import styles from './LandingStyles';

class Landing extends React.Component {

  static propTypes = {
    classes: PropTypes.object.isRequired
  };

  render = () => {
    const { classes } = this.props;
    return (
      <div>
        <div className={classes.use}>
          <h1 className={classes.landingTitle}>
            Use our digital service for
          </h1>
          <div className={classes.useGrid}>
            <div className={classes.useElement}>
              <img src="/svg/001-bank.svg" alt="" className={classes.useImage}/>
              <div className={classes.useTitle}> Create an account</div>
              <div className={classes.useDescription}> 
                You can create an account in the currency you choose.
              </div>
            </div>
            <div className={classes.useElement}>
              <img src="/svg/004-budget.svg" alt="" className={classes.useImage}/>
              <div className={classes.useTitle}> Trust us with your money </div>
              <div className={classes.useDescription}> 
                You can transfer your money to your account and it will be safe with us.
              </div>
            </div>
            <div className={classes.useElement}>
              <img src="/svg/014-dollar-bill.svg" alt="" className={classes.useImage}/>
              <div className={classes.useTitle}> Transfer your money </div>
              <div className={classes.useDescription}> 
                You can transfer your money to other people in a secure way.
              </div>
            </div>
          </div>
        </div>

        <div className={classes.howItWorks}>
          <h2 className={classes.howItWorksTitle}>
            How it all works?
          </h2>
          <div className={classes.howItWorksParagraph}>
            Our application it is very simple to use. You create an account and then you are ready to add, withdraw
            and transfer the money from your account. We are very focused on simplicity so we made this app for you
            to manage all these operations in the simplest way possible.
          </div>
        </div>
      </div>
    );
  };
}

export default compose(withStyles(styles))(Landing);
