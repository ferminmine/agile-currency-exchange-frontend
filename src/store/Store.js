import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import { createBrowserHistory } from 'history';
import thunk from 'redux-thunk';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import hardSet from 'redux-persist/lib/stateReconciler/hardSet';
import userReducer from '../modules/user/UserReducer';
import currenciesReducer from '../modules/currencies/CurrenciesReducer';
import accountReducer from '../modules/account/AccountReducer';

export const history = createBrowserHistory();

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['router'],
  stateReconciler: hardSet
};

const allReducers = combineReducers({
  user: userReducer,
  currencies: currenciesReducer,
  account: accountReducer,
  router: connectRouter(history)
});

export const ROOT_REDUCER_ACTION_TYPES = {
  RESET_APP: 'RESET_APP'
};
export const resetApp = () => ({ type: ROOT_REDUCER_ACTION_TYPES.RESET_APP });

const rootReducer = (state, action) => {
  if (action.type === 'RESET_APP') {
    state = undefined;
  }

  return allReducers(state, action);
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  persistedReducer,
  composeEnhancer(applyMiddleware(routerMiddleware(history), thunk))
);
export default store;
export const persistor = persistStore(store);
