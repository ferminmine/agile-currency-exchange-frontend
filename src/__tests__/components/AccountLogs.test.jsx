import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import Landing from '../../modules/landing/Landing';
import store from '../../store/Store';

describe('Landing Component Test', () => {
  it('renders without crashing', () => {
    mount(
      <Provider store={store}>
        <Landing />
      </Provider>
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Landing />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
