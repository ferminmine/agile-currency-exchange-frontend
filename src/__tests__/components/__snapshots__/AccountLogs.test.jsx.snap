// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component Test matches the snapshot 1`] = `
<div>
  <div
    className="Landing-use-0-1-1"
  >
    <h1
      className="Landing-landingTitle-0-1-2"
    >
      Use our digital service for
    </h1>
    <div
      className="Landing-useGrid-0-1-3"
    >
      <div
        className="Landing-useElement-0-1-5"
      >
        <img
          alt=""
          className="Landing-useImage-0-1-4"
          src="/svg/001-bank.svg"
        />
        <div
          className="Landing-useTitle-0-1-6"
        >
           Create an account
        </div>
        <div
          className="Landing-useDescription-0-1-7"
        >
          You can create an account in the currency you choose.
        </div>
      </div>
      <div
        className="Landing-useElement-0-1-5"
      >
        <img
          alt=""
          className="Landing-useImage-0-1-4"
          src="/svg/004-budget.svg"
        />
        <div
          className="Landing-useTitle-0-1-6"
        >
           Trust us with your money 
        </div>
        <div
          className="Landing-useDescription-0-1-7"
        >
          You can transfer your money to your account and it will be safe with us.
        </div>
      </div>
      <div
        className="Landing-useElement-0-1-5"
      >
        <img
          alt=""
          className="Landing-useImage-0-1-4"
          src="/svg/014-dollar-bill.svg"
        />
        <div
          className="Landing-useTitle-0-1-6"
        >
           Transfer your money 
        </div>
        <div
          className="Landing-useDescription-0-1-7"
        >
          You can transfer your money to other people in a secure way.
        </div>
      </div>
    </div>
  </div>
  <div
    className="Landing-howItWorks-0-1-8"
  >
    <h2
      className="Landing-howItWorksTitle-0-1-9"
    >
      How it all works?
    </h2>
    <div
      className="Landing-howItWorksParagraph-0-1-10"
    >
      Our application it is very simple to use. You create an account and then you are ready to add, withdraw and transfer the money from your account. We are very focused on simplicity so we made this app for you to manage all these operations in the simplest way possible.
    </div>
  </div>
</div>
`;
